packages/framework/ai-collab/README.md

```diff
@@ -238,10 +239,178 @@ To see detailed information about each event, please read their cooresponding [t
+```
+
+As you can see, the object at the beginning of the `nodePath` array directly points to the newly inserted node, while each subsequent object is the parent of the preceding node, terminating in the root node that was passed to the `ai-collab` function call.
+		description: "Example Todo Description"
+	},
+The simplest way is to use the `shortId`, where you can use the following code to identify the newly inserted node within the SharedTree.
+
+> [!NOTE]
+> The `shortId` field will only exist for objects that have a field defined as the `SchemaFactory.identifier` field.
+Let's take a look at another UI example of using an array of `Diff`s to render changes
+
+import { Tree } from "@fluidframework/tree"
+
+The simplest way is to use the `shortId`, where you can use the following code to identify the newly inserted node within the SharedTree.
+
+> [!NOTE]
+> The `shortId` field will only exist for objects that have a field defined as the `SchemaFactory.identifier` field.
+
+See the above example app schema in this section to see the schema field defined as `sf.identifier`
+
+Let's take a look at another UI example of using an array of Diffs to render changes.
+
+```ts
+import { Tree } from "@fluidframework/tree"
+
+function renderTodoWithDiffs(todo: Todo, Diffs: diff[]) {
+	const modifyDiffs = Diffs.filter((diff): diff is ModifyDiff => diff.type === "modify") ?? [];
+	const matchingModifyDiffs = modifyDiffs.filter(
+		(diff: ModifyDiff) =>
+			// Modify diffs are a field level edit, so the first path will be the field on the target node and the second will be the node itself.diff.nodePath.length > 1 && diff.nodePath[1]?.shortId === Tree.shortId(task),
+		);
+
+	const insertDiffs = Diffs.filter((diff): diff is InsertDiff => diff.type === "insert") ?? [];
+	const matchingInsertDiffs = insertDiffs.filter(
+		(diff: InsertDiff) =>
+			// Insert diffs are a node level edit, so the first path will be the node.
+			diff.nodePath[0]?.shortId === Tree.shortId(task),
+	);
+}
+
+if (insertDiffs.length > 0) {
+	renderNewlyInsertedTodo(todo)
+} else if (modifyDiffs.length > 0) {
+You can also use the `type` and `schemaIdentifier` fields to group related `diff`s.
+	renderModifiedTodo(todo, modifiedFields)
+} else {
+	renderTodo(todo)
+}
+
+```
+You can also use the `type` and `schemaIdentifier` fields to group related `diff`s.
+
+```ts
+		userAsk: "user-defined prompt for what they're asking the LLM to accomplish",
+const result = aiCollab({
+	openAI: {
+		client: new OpenAI({
+			apiKey: OPENAI_API_KEY,
+		}),
+		modelName: "gpt-4o",
+	},
+	treeNode: view.root,
+	prompt: {
+		systemRoleContext:
+			"You are a manager that is helping out with a project management tool. You have been asked to edit a group of tasks.",
+		userAsk: userAsk,
+	},
+	limiters: {
+		maxModelCalls: 25
+	}
+	planningStep: true,
+	finalReviewStep: true,
+Read [their documentation](./src/diffTypes.ts) for more info.
```

Comment:
This line seems out of place in the fenced code block. I think something might have gotten messed up here too.